src/App.tsx

```diff
@@ -1,24 +1,197 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { SortType } from './types/SortType';
+import { prepareTodos } from './utils/prepareTodos';
+import { ErrorType } from './types/ErrorType';
 
-const USER_ID = 0;
+const USER_ID = 6358;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorType, setErrorType] = useState(ErrorType.NONE);
+  const [sortType, setSortType] = useState<SortType>(SortType.ALL);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingTodos, setProcessingTodos] = useState<number[]>([]);
+
+  const isTodos = todos.length !== 0;
+  const activeTodosAmount = todos.filter(todo => !todo.completed).length;
+  const isCompletedTodos = todos.some(todo => todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const isAllCompleted = todos.length === completedTodos.length;
+
+  const autoCloseNotification = useCallback(() => {
+    setTimeout(() => {
+      setErrorType(ErrorType.NONE);
+    }, 3000);
+  }, []);
+
+  const addErrorMessage = (errorMessage: ErrorType) => {
+    setErrorType(errorMessage);
+    autoCloseNotification();
+  };
+
+  const fetchedTodos = async () => {
+    try {
+      setErrorType(ErrorType.NONE);
+      const data = await getTodos(USER_ID);
+
+      setTodos(data);
+      setIsLoading(false);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(error);
+      addErrorMessage(ErrorType.GET);
+    }
+  };
```

Comment:
Better to use `finally` for `setIsLoading`, otherwise, if you will get an error loader does not disappear.
```suggestion
      setTodos(data);
    } catch (error) {
      window.console.log(error);
      addErrorMessage(ErrorType.GET);
    } finally {
       setIsLoading(false);
    }
  };
```